lib/src/component_declaration/component_base_2.dart

```diff
@@ -664,12 +660,12 @@ mixin UiStatefulMixin2<TProps extends UiProps, TState extends UiState> on UiComp
 
   /// Returns a typed state object backed by a Map.
   ///
-  /// Convenient for use with [getInitialState] and [setState].
+  /// Convenient for use with [initialState] and [setState].
   TState newState() => typedStateFactoryJs(JsBackedMap());
 
   @override
-  void setStateWithUpdater(covariant Map Function(TState prevState, TProps props) updater, [Function() callback]) {
-    final bridge = Component2Bridge.forComponent(this) as UiComponent2BridgeImpl;
+  void setStateWithUpdater(covariant Map? Function(TState prevState, TProps props) updater, [Function()? callback]) {
```

Comment:
Yeah, React JS allows returning null (see TS typing [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/398d563b05acda2c203e5144aefa2cac9500ae33/types/react/index.d.ts#L530)), which IIRC results in no state update occurring.

Reply:
Might be worth adding a comment to that effect since at first glance - its like... uh, wait - why/how could state ever be null? When in reality - null is more akin to a special case handled by react to prevent a state update - and it doesn't actually mutate state to be null.

Reply:
I think that's fair, but this typing and the doc comment is inherited from the super-class, which is in react-dart, so I don't think there's anything actionable for this over_react PR.